Automattic/wp-calypso
# Full Site Editing: Refactor the Template block UI

#### Changes proposed in this Pull Request

Requires https://github.com/Automattic/themes/pull/1262

Refactor the Template block style, and changes the block selection behaviour.

* Drop all the unnecessary rules that were left over during the months of development back and forth.
* Hide all the block's UI component that aren't used by FSE (breadcrumb, toolbar, borders).
* Move the theme classes (e.g. `.site-header`) from the block most external wrapper, to one more inside. This should simplify the overrides, as now the theme styles are only applied to the rendered part of the block, and not to its wrapper too.
* And much else! (I'll add to this list later)

#### SCREENSHOTS GALORE!

<details><summary>Show</summary>
<p>

**Clean Slate**
Note: the colophon positioning is off compared to how it's supposed to look in the theme (which is a bit awkward, what with calculated margins that attempt to almost center position it, instead of just `text-align: center` it and call it a day. tl;dr it's technically `text-align: left`, but looks more centered)

| Where | Current | Refactor |
| - | - | - |
| Header | ![og-clean-header](https://user-images.githubusercontent.com/2070010/63279932-ca3c8d80-c2a1-11e9-9156-d692aa609430.png) | ![refactor-clean-header](https://user-images.githubusercontent.com/2070010/63280030-fce68600-c2a1-11e9-96a1-9e05dfa3b90c.png) |
| Footer | ![og-clean-footer](https://user-images.githubusercontent.com/2070010/63279947-d1fc3200-c2a1-11e9-8f2a-eda227b9adc7.png) | ![refactor-clean-footer](https://user-images.githubusercontent.com/2070010/63280037-02dc6700-c2a2-11e9-823c-d2fc9b92cf44.png) |
| Page | ![og-clean-page](https://user-images.githubusercontent.com/2070010/63279993-e9d3b600-c2a1-11e9-8e71-1ebb9d934038.png) | ![refactor-clean-page](https://user-images.githubusercontent.com/2070010/63280052-096ade80-c2a2-11e9-8c5b-6033b5702494.png) |
| Front End | ![og-clean-front](https://user-images.githubusercontent.com/2070010/63280003-ef310080-c2a1-11e9-8dc6-06f45204d892.png) | ![refactor-clean-front](https://user-images.githubusercontent.com/2070010/63280065-1091ec80-c2a2-11e9-905a-9044b4575e9a.png) |
| Theme Demo | ![theme-footer](https://user-images.githubusercontent.com/2070010/63280852-76cb3f00-c2a3-11e9-8f40-f80d47381407.png) | |

**Header with Logo**
Note: the header elements spacings are off in both current and refactor, compared to how the theme is supposed to look.

| Where | Current | Refactor |
| - | - | - |
| Header | ![og-logo-header](https://user-images.githubusercontent.com/2070010/63280189-420ab800-c2a2-11e9-9aaf-4fa86562997f.png) | ![refactor-logo-header](https://user-images.githubusercontent.com/2070010/63280232-5cdd2c80-c2a2-11e9-9f8f-ff43c5e454e5.png) |
| Page | ![og-logo-page](https://user-images.githubusercontent.com/2070010/63280199-48009900-c2a2-11e9-836b-cf0d8945f1a2.png) | ![refactor-logo-page](https://user-images.githubusercontent.com/2070010/63280273-6cf50c00-c2a2-11e9-9e1d-ec05d8e07fd3.png) |
| Front End | ![og-logo-front](https://user-images.githubusercontent.com/2070010/63280210-4cc54d00-c2a2-11e9-8cd5-983a3ae6d6ba.png) | ![refactor-logo-front](https://user-images.githubusercontent.com/2070010/63280279-72eaed00-c2a2-11e9-94f5-73142dff63eb.png) |
| Theme Demo | ![theme-header](https://user-images.githubusercontent.com/2070010/63280882-8b0f3c00-c2a3-11e9-8cfd-75a42570974e.png) | |


**Header with Full Width Image**
Note: in both current and refactor, on the front end, the image has a negative right margin that triggers the horizontal scrollbar.

| Where | Current | Refactor |
| - | - | - |
| Header | ![og-image-header](https://user-images.githubusercontent.com/2070010/63280356-944bd900-c2a2-11e9-838c-42a4a4c2830f.png) | ![refactor-image-header](https://user-images.githubusercontent.com/2070010/63280468-be9d9680-c2a2-11e9-9dc1-bcd3a128b23f.png) |
| Page | ![og-image-page](https://user-images.githubusercontent.com/2070010/63280410-a3328b80-c2a2-11e9-99fd-0b1415b7d3ae.png) | ![refactor-image-page](https://user-images.githubusercontent.com/2070010/63280476-c52c0e00-c2a2-11e9-9bc4-5951c03c2e68.png) |
| Front End | ![og-image-front](https://user-images.githubusercontent.com/2070010/63280438-b04f7a80-c2a2-11e9-9261-c33f75526371.png) | ![refactor-image-front](https://user-images.githubusercontent.com/2070010/63280485-c9f0c200-c2a2-11e9-9c4b-3be32f10c390.png) |

**Footer Alignment**
Note: imho this has a very bad UX.
Blocks have an alignment option, but then some of them are forced with a predetermined alignment. Imho, we should not force any alignment via CSS, but instead letting people do their thing with the editor.

| Where | Current | Refactor |
| - | - | - |
| Footer | ![og-align-footer](https://user-images.githubusercontent.com/2070010/63281230-299b9d00-c2a4-11e9-9dc2-df77c9959af8.png) | ![refactor-align-footer](https://user-images.githubusercontent.com/2070010/63281271-3ae4a980-c2a4-11e9-9a14-cd7c915e44a9.png) |
| Page | ![og-align-page](https://user-images.githubusercontent.com/2070010/63281237-2e605100-c2a4-11e9-9ef8-8074fbf15a23.png) | ![refactor-align-page](https://user-images.githubusercontent.com/2070010/63281281-3f10c700-c2a4-11e9-8877-b6ba738eb8a4.png) |
| Front End | ![og-align-front](https://user-images.githubusercontent.com/2070010/63281256-328c6e80-c2a4-11e9-8480-dff094ec3341.png) | ![refactor-align-front](https://user-images.githubusercontent.com/2070010/63281290-43d57b00-c2a4-11e9-8e40-eebac2e70f08.png) |

**Selecting a Template Doesn't Hide the Sidebar Anymore**
Note: the sidebar stays open (if already open), and the "Block" tab gets hidden.

| Current | Refactor
| - | - |
| ![og-sidebar-page](https://user-images.githubusercontent.com/2070010/63281380-77180a00-c2a4-11e9-8935-a369ebb7a43b.png) | ![refactor-sidebar-page](https://user-images.githubusercontent.com/2070010/63281397-84cd8f80-c2a4-11e9-8ff5-bd0f64a30445.png) |


</p>
</details>

#### Testing instructions



* Build the Modern Business theme with https://github.com/Automattic/themes/pull/1262.
* On an FSE site, edit a page.
* Try poking the header and footer template parts, and make sure that:
  * The Template block overlay is properly aligned to the editor frame.
  * The overlay retains the same background color and blur filter as before.
  * Selecting (clicking) a Template block does not close the sidebar (if it was open), but simply hides its "Block" tab.
  * Everything looks fine!

Fixes https://github.com/Automattic/wp-calypso/issues/35378
